main.go

```diff
@@ -89,47 +93,86 @@ func Run(sigs chan os.Signal) error {
 			log.Printf("HTTP server shut down successfully")
 
 			return nil
-		default:
-			time.Sleep(1 * time.Second)
-			// S3 GetObject
-			getStart := time.Now()
-			_, err = s3Client.GetObject(context.Background(), &s3.GetObjectInput{
-				Bucket: &s3Bucket,
-				Key:    &s3Key,
-			})
-			getDuration := time.Since(getStart).Seconds()
-			if err != nil {
-				log.Printf("failed to get object, %v", err)
-				requestDuration.WithLabelValues("S3", "GetObject", "Failure").Observe(getDuration)
-			} else {
-				requestDuration.WithLabelValues("S3", "GetObject", "Success").Observe(getDuration)
+		case <-time.After(1 * time.Second):
+			select {
+			case checks <- struct{}{}:
+			default:
 			}
+		}
+	}
+}
 
-			// DynamoDB Scan
-			dynamoStart := time.Now()
-			_, err = dynamoClient.Scan(context.Background(), &dynamodb.ScanInput{
-				TableName: &dynamodbTable,
-			})
-			dynamoDuration := time.Since(dynamoStart).Seconds()
-			if err != nil {
-				log.Printf("failed to get item, %v", err)
-				requestDuration.WithLabelValues("DynamoDB", "Scan", "Failure").Observe(dynamoDuration)
-			} else {
-				requestDuration.WithLabelValues("DynamoDB", "Scan", "Success").Observe(dynamoDuration)
-			}
+type checker struct {
+	s3Client     *s3.Client
+	dynamoClient *dynamodb.Client
+	sqsCleint    *sqs.Client
```

Comment:
`c.sqsCleint` -> `c.sqsClient`?

*It's referenced in multiple places so this PR may not be the best place to change it.

Reply:
Thanks! Will fix in another PR!

Reply:
See 9688d9939931e994b47f81019f2907812bbc5042